### PR TITLE
Support for Gipfeli compression in the PushDeserializer

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

**Comment by pleroy** [comment]
retest this please

**Comment by pleroy** [comment]
retest this please

**Comment by pleroy** [comment]
retest this please

File: base/push_deserializer_test.cpp [hunk 1]
```diff
@@ -149,6 +189,75 @@ TEST_F(PushDeserializerTest, Stream) {
   EXPECT_EQ(23, stream_.ByteCount());
 }
 
+TEST_F(PushDeserializerTest, DeserializationGipfeli) {
+  Compressor* compressor = google::compression::NewGipfeliCompressor();
```
Comment (eggrobin) [inline base/push_deserializer_test.cpp]: "must be deleted after use", says the comment. We should have a MakeGipfeliCompressor wrapper that returns std::unique_ptr (probably same comment for the decompression test).

Comment (pleroy) [reply base/push_deserializer_test.cpp]: Done.

**Pull request merged**